### ISSUE TITLE
`[no-inject-outside-ctor]` new rule

### ISSUE THREAD

**Issue by Lonli-Lokli** [angular-eslint/angular-eslint#1400]
**Description and reproduction of the issue**
Currently angular has powerful inject function, which supposed to replace ctor. Unfortunately sometimes you might use it outside of constructor context which will fsil

**Comment by itotallyrock** [comment]
As far as only allowing in the constructor, I think property-initializers would need to be considered valid as well

```ts
class Foo {
    private readonly bar = inject(Bar);
}
```

Might be also worth checking that the class is also `@Injectable` in the first place.

---

Beyond specifics, it would also be nice to have similar rules for `effect` and `toSignal` which both require an injection context.

**Comment by reduckted** [comment]
This could be tricky to do correctly because there are some places outside of constructors that run in an injection context, which allows you to use `inject`.

- [HTTP interceptors](https://angular.dev/guide/http/interceptors#dependency-injection-in-interceptors).
- [Route guards](https://github.com/angular/angular/blob/51b8ff23cefb5112937dec9727a5b5d6e913aae6/packages/router/src/operators/check_guards.ts#L173)
- [Resolvers](https://github.com/angular/angular/blob/51b8ff23cefb5112937dec9727a5b5d6e913aae6/packages/router/src/operators/resolve_data.ts#L9)
- [App initializers](https://angular.dev/api/core/provideAppInitializer#description)
- [Environment initializers](https://angular.dev/api/core/provideEnvironmentInitializer#description)
- [Platform initializers](https://angular.dev/api/core/providePlatformInitializer#description)

There's potentially other places that I've missed. Unfortunately, a lot of those things are just normal functions, which would make it difficult to reliably detect that they are running in an injection context.

It's a shame, because this would be a nice rule. ☹
